relay-pattern/src/lib.rs

```diff
@@ -1427,6 +1462,9 @@ mod tests {
         assert_pattern!("{foo,bar}", NOT "BAR");
         assert_pattern!("{foo,bar}", NOT "fooo");
         assert_pattern!("{foo,bar}", NOT "baar");
+        assert_pattern!("{foo,bar,}baz", "foobaz");
+        assert_pattern!("{foo,bar,}baz", "barbaz");
+        assert_pattern!("{foo,bar,}baz", "baz");
```

Comment:
Should we add tests for `{foo,,bar}baz` and `{,foo,bar}baz`.

Reply:
Glad you mentioned this because `{,foo,bar}` currently fails.

Reply:
Should have a test for all added branches:

- `{foo*bar}`
- `{foo*bar,}`
- `{foo*bar,baz,}`
- `{foo*bar,baz}`
- `{,,,}`

Some of them probably already exist.